kore/src/Kore/Step/Step.hs

```diff
@@ -224,8 +229,26 @@ unifyRule
     let
         RulePattern { requires = ruleRequires } = rule'
         requires' = Predicate.fromPredicate ruleRequires
-    unification' <- Substitution.normalizeExcept (unification <> requires')
+    unification' <- removeRedundancy unification requires'
     return (rule' `Conditional.withCondition` unification')
+  where
+    removeRedundancy unificationSolution requires = do
```

Comment:
The name `removeRedundancy` is unclear about what is redundant. Maybe, `removeRedundantRequires`?

Reply:
Actually, wouldn't `addRequiresUnlessRedundant` be a better name? This does not remove anything...